diff --git a/debian/rules b/debian/rules
index e26fb54..799c9ef 100755
--- a/debian/rules
+++ b/debian/rules
@@ -50,6 +50,7 @@ confflags = --host=$(DEB_HOST_GNU_TYPE) \
                        --mandir=\$${prefix}/share/man \
                        --localstatedir=/var --sysconfdir=/etc \
                        --with-perl-bindings="INSTALLDIRS=vendor INSTALL_BASE=" \
+                       --with-libdpdk=/usr \
                        --without-libstatgrab \
                        --without-included-ltdl \
                        --disable-static \
@@ -258,7 +259,7 @@ binary-indep: install-indep
        dh_testdir
        dh_testroot
        dh_installchangelogs -i ChangeLog
-       dh_installdocs -A -i debian/README.Debian AUTHORS README TODO
+       dh_installdocs -A -i debian/README.Debian AUTHORS README
        dh_installexamples -i contrib/examples/myplugin.c \
                contrib/examples/MyPlugin.pm
        dh_compress -i -Xexamples/
@@ -272,7 +273,7 @@ binary-arch: build install-arch
        dh_testdir
        dh_testroot
        dh_installchangelogs -a ChangeLog
-       dh_installdocs -A -a debian/README.Debian AUTHORS README TODO
+       dh_installdocs -A -a debian/README.Debian AUTHORS README
        dh_installdocs -a debian/NEWS.Debian debian/README.Debian.plugins
        dh_installexamples -a contrib/collectd2html.pl contrib/collection.cgi \
                contrib/collection3/ contrib/php-collection/ \
@@ -305,7 +306,7 @@ binary-arch: build install-arch
        dh_shlibdeps -a -Ncollectd-core -Ncollectd
        dpkg-shlibdeps -Tdebian/collectd.substvars \
                -dDepends debian/collectd-core/usr/lib/collectd/rrdtool.so
-       dpkg-shlibdeps -Tdebian/collectd-core.substvars \
+       dpkg-shlibdeps --ignore-missing-info -Tdebian/collectd-core.substvars \
                -dDepends debian/collectd-core/usr/sbin/* \
                -dSuggests debian/collectd-core/usr/lib/collectd/*.so
        grep shlibs:Suggests debian/collectd-core.substvars \

