diff --git a/src/intel_rdt.c b/src/intel_rdt.c
index 6beac7b..2656cb9 100644
--- a/src/intel_rdt.c
+++ b/src/intel_rdt.c
@@ -36,6 +36,12 @@
 #define RDT_MAX_SOCKET_CORES 64
 #define RDT_MAX_CORES (RDT_MAX_SOCKET_CORES * RDT_MAX_SOCKETS)

+typedef enum {
+  UNKOWN = 0,
+  CONFIGURATION,
+  CONFIGURATION_ERROR,
+} rdt_config_status;
+
 struct rdt_core_group_s {
   char *desc;
   size_t num_cores;
@@ -56,6 +62,8 @@ typedef struct rdt_ctx_s rdt_ctx_t;

 static rdt_ctx_t *g_rdt = NULL;

+static rdt_config_status g_state = UNKOWN;
+
 static int isdup(const uint64_t *nums, size_t size, uint64_t val) {
   for (size_t i = 0; i < size; i++)
     if (nums[i] == val)
@@ -525,10 +533,17 @@ rdt_preinit_error1:

 static int rdt_config(oconfig_item_t *ci) {
   int ret = 0;
+  g_state = CONFIGURATION;

   ret = rdt_preinit();
-  if (ret != 0)
-    return ret;
+  if (ret != 0) {
+    g_state = CONFIGURATION_ERROR;
+    /* if we return -1 at this point collectd
+      reports a failure in configuration and
+      aborts
+    */
+    goto exit;
+  }

:

+static rdt_config_status g_state = UNKOWN;
+
 static int isdup(const uint64_t *nums, size_t size, uint64_t val) {
   for (size_t i = 0; i < size; i++)
     if (nums[i] == val)
@@ -525,10 +533,17 @@ rdt_preinit_error1:

 static int rdt_config(oconfig_item_t *ci) {
   int ret = 0;
+  g_state = CONFIGURATION;

   ret = rdt_preinit();
-  if (ret != 0)
-    return ret;
+  if (ret != 0) {
+    g_state = CONFIGURATION_ERROR;
+    /* if we return -1 at this point collectd
+      reports a failure in configuration and
+      aborts
+    */
+    goto exit;
+  }

   for (int i = 0; i < ci->children_num; i++) {
     oconfig_item_t *child = ci->children + i;
@@ -536,8 +551,14 @@ static int rdt_config(oconfig_item_t *ci) {
     if (strcasecmp("Cores", child->key) == 0) {

       ret = rdt_config_cgroups(child);
-      if (ret != 0)
-        return ret;
+      if (ret != 0) {
+        g_state = CONFIGURATION_ERROR;
+        /* if we return -1 at this point collectd
+           reports a failure in configuration and
+           aborts
+         */
+        goto exit;
+      }

 #if COLLECT_DEBUG
       rdt_dump_cgroups();
@@ -548,6 +569,7 @@ static int rdt_config(oconfig_item_t *ci) {
     }
   }

+exit:
   return (0);
 }

@@ -630,6 +652,9 @@ static int rdt_read(__attribute__((unused)) user_data_t *ud) {
 static int rdt_init(void) {
   int ret;

+  if(g_state == CONFIGURATION_ERROR)
+    return (-1);
+
   ret = rdt_preinit();
   if (ret != 0)
     return ret;
